examples/animation/animation_graph.rs

```diff
@@ -0,0 +1,538 @@
+const WEIGHT_ADJUST_SPEED: f32 = 0.01;
+
+/// The help text in the upper left corner.
+static HELP_TEXT: &str = "Click and drag an animation clip node to change its weight";
+
+/// The node widgets in the UI.
+static NODE_TYPES: [NodeType; 5] = [
+    NodeType::Clip(ClipNodeText::new("Idle", 0)),
+    NodeType::Clip(ClipNodeText::new("Walk", 1)),
+    NodeType::Blend("Root"),
+    NodeType::Blend("Blend\n0.5"),
+    NodeType::Clip(ClipNodeText::new("Run", 2)),
+];
+
+/// The positions of the node widgets in the UI.
+///
+/// These are in the same order as [`NODE_TYPES`] above.
+static NODE_RECTS: [NodeRect; 5] = [
+    NodeRect::new(10.00, 10.00, 97.64, 48.41),
+    NodeRect::new(10.00, 78.41, 97.64, 48.41),
+    NodeRect::new(286.08, 78.41, 97.64, 48.41),
+    NodeRect::new(148.04, 44.20, 97.64, 48.41),
+    NodeRect::new(10.00, 146.82, 97.64, 48.41),
+];
+
+/// The positions of the horizontal lines in the UI.
+static HORIZONTAL_LINES: [Line; 6] = [
+    Line::new(107.64, 34.21, 20.20),
+    Line::new(107.64, 102.61, 20.20),
+    Line::new(107.64, 171.02, 158.24),
+    Line::new(127.84, 68.41, 20.20),
+    Line::new(245.68, 68.41, 20.20),
+    Line::new(265.88, 102.61, 20.20),
+];
+
+/// The positions of the vertical lines in the UI.
+static VERTICAL_LINES: [Line; 2] = [
+    Line::new(127.83, 34.21, 68.40),
+    Line::new(265.88, 68.41, 102.61),
+];
+
+/// Initializes the app.
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins.set(WindowPlugin {
+            primary_window: Some(Window {
+                title: "Bevy Animation Graph Example".into(),
+                ..default()
+            }),
+            ..default()
+        }))
+        .add_systems(Startup, setup)
+        .add_systems(PreUpdate, init_animations)
+        .add_systems(
+            Update,
+            (handle_weight_drag, update_ui, sync_weights).chain(),
+        )
+        .init_resource::<AnimationWeights>()
+        .init_resource::<DragState>()
+        .insert_resource(AmbientLight {
+            color: Srgba(WHITE),
+            brightness: 100.0,
+        })
+        .run();
+}
+
+fn setup(
+    mut commands: Commands,
+    mut asset_server: ResMut<AssetServer>,
+    mut animation_graphs: ResMut<Assets<AnimationGraph>>,
+) {
+    // Create the assets.
+    setup_assets(&mut commands, &mut asset_server, &mut animation_graphs);
+
+    // Create the scene.
+    setup_scene(&mut commands, &mut asset_server);
+    setup_camera_and_light(&mut commands);
+
+    // Create the UI.
+    setup_help_text(&mut commands, &mut asset_server);
+    setup_node_rects(&mut commands, &mut asset_server);
+    setup_node_lines(&mut commands);
+}
+
+/// Creates the assets, including the animation graph.
+fn setup_assets(
+    commands: &mut Commands,
+    asset_server: &mut AssetServer,
+    animation_graphs: &mut Assets<AnimationGraph>,
+) {
+    // Create the nodes.
+    let mut animation_graph = AnimationGraph::new();
```

Comment:
This step should be called out with a comment, and ideally we would use a non-empty animation graph.

Reply:
I'm confused. This isn't an empty animation graph, as we add to it.